frontend/src/stimulus/controllers/dynamic/work-packages/activities-tab/stems.controller.ts

```diff
@@ -0,0 +1,81 @@
+/*
+ * -- copyright
+ * OpenProject is an open source project management software.
+ * Copyright (C) 2023 the OpenProject GmbH
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License version 3.
+ *
+ * OpenProject is a fork of ChiliProject, which is a fork of Redmine. The copyright follows:
+ * Copyright (C) 2006-2013 Jean-Philippe Lang
+ * Copyright (C) 2010-2013 the ChiliProject Team
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * See COPYRIGHT and LICENSE files for more details.
+ * ++
+ */
+
+import { Controller } from '@hotwired/stimulus';
+import { withIndexOutletMixin } from './mixins/with-index-outlet';
+
+export default class StemsController extends withIndexOutletMixin(Controller) {
```

Comment:
This inheritance is not really obvious to me. It takes a moment to understand.

In the end it's a clever way of injecting functionality so I think it works well. But I cannot avoid wondering if there's a way we don't depend on making it a function. Or if we could simplify it by not needing to pass down the `Controller`.

In any case, this is just an observation, as I don't have a better idea on how to do it~

Reply:
Good eye, see https://github.com/opf/openproject/pull/19669 that attempts to simplify this 🙂